### ISSUE TITLE
[Issue]: Unable to build, Unknown CMake command "pybind11_add_module"

### ISSUE THREAD

**Issue by RandUser123sa** [ROCm/aotriton#28]
### Problem Description

Hi,

unable to build the project. I download last release and following documentation do:

mkdir build && cd build
cmake .. -DCMAKE_INSTALL_PREFIX=./install_dir -DCMAKE_BUILD_TYPE=Release

I receive the error:

# cmake .. -DCMAKE_INSTALL_PREFIX=./install_dir -DCMAKE_BUILD_TYPE=Release
-- The CXX compiler identification is GNU 14.1.0
-- The C compiler identification is GNU 14.1.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
CMake Error at CMakeLists.txt:8 (add_subdirectory):
  The source directory

    /root/tour/trit/aotriton-0.5b/third_party/pybind11

  does not contain a CMakeLists.txt file.

-- Found Python3: /root/newt/bin/python3.11 (found version "3.11.9") found components: Interpreter
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE
CMake Error at CMakeLists.txt:46 (get_property):
  get_property could not find TARGET zstd::libzstd_static.  Perhaps it has
  not yet been created.

-- ZSTD_TARGET zstd::libzstd_static
-- get_property AOTRITON_ZSTD_INCLUDE 
VENV_SITE /root/.local/lib/python3.11/site-packages

CMAKE_SOURCE_DIR /root/tour/trit/aotriton-0.5b
CMAKE_CURRENT_LIST_DIR /root/tour/trit/aotriton-0.5b/v2src
CMAKE_CURRENT_BINARY_DIR /root/tour/trit/aotriton-0.5b/build/v2src
-- AOTRITON_ZSTD_INCLUDE 
-- AOTRITON_SHIM_FLAGS --archive_only;
-- AOTRITON_EXTRA_COMPILER_OPTIONS -I/opt/rocm/include/  -O3 -DNDEBUG
CMAKE_INSTALL_INCLUDEDIR include
CMake Error at bindings/CMakeLists.txt:6 (pybind11_add_module):
  Unknown CMake command "pybind11_add_module".

-- Configuring incomplete, errors occurred!

Any suggestion ?

Thanks

### Operating System

Slackware 15.0

### CPU

AMD Ryzen 7 3800X 8-Core Processor

### GPU

AMD Radeon RX 7900 XTX

### ROCm Version

ROCm 6.1.0

**Comment by xinyazhang** [comment]
Oh, you need to clone the repo recursively. If you have already cloned one, use the following command to update all submodules.
```
git submodule sync
git submodule update --init --recursive --force
```

**Comment by lhl** [comment]
@xinyazhang hi, ran into the issue trying to build today - would you mind updating the README or would you rather a PR?

**Comment by xinyazhang** [comment]
> @xinyazhang hi, ran into the issue trying to build today - would you mind updating the README or would you rather a PR?

Can you confirm the command above works? Otherwise we are not confident this is the proper fix to include into the README.md.

Also It's much appreciated if you can file a PR (to main branch) since the the main branch is locked and requires reviews (as owner technically I can bypass this but I don't want to be the sole exception)

**Comment by RandUser123sa** [comment]
Probably will work if some one use GIT version, but still does not compile if I download the Release version. I tested on latest aotriton-0.7preview2 downloaded from here: https://github.com/ROCm/aotriton/releases

```
-- Detecting C compile features - done
CMake Error at CMakeLists.txt:8 (add_subdirectory):
  The source directory

    /root/tour/trit/aotriton-0.7preview2/third_party/pybind11

  does not contain a CMakeLists.txt file.


-- Found Python3: /usr/bin/python3.11 (found version "3.11.9") found components: Interpreter

```

and bellow is the error:

```
tride_rz', 'stride_rm'}): ['u64:16']}
self.FEAT_CHOICES={frozenset({'stride_rn'}): [1]}
For kernel flash.bwd_preprocess, Architecture gfx90a was not found in tuning database, using dummy one instead
For kernel flash.bwd_preprocess, Architecture gfx942 was not found in tuning database, using dummy one instead
For kernel flash.bwd_preprocess, Architecture gfx1100 was not found in tuning database, using dummy one instead
For kernel flash.bwd_preprocess_varlen, Architecture gfx90a was not found in tuning database, using dummy one instead
For kernel flash.bwd_preprocess_varlen, Architecture gfx942 was not found in tuning database, using dummy one instead
For kernel flash.bwd_preprocess_varlen, Architecture gfx1100 was not found in tuning database, using dummy one instead
For kernel flash.debug_fill_dropout_rng, Architecture gfx90a was not found in tuning database, using dummy one instead
For kernel flash.debug_fill_dropout_rng, Architecture gfx942 was not found in tuning database, using dummy one instead
For kernel flash.debug_fill_dropout_rng, Architecture gfx1100 was not found in tuning database, using dummy one instead
Skip rpath=PosixPath('template/autotune_table_entry.cc')
Skip rpath=PosixPath('template/shim.cc')
CMAKE_INSTALL_INCLUDEDIR include
CMake Error at bindings/CMakeLists.txt:6 (pybind11_add_module):
  Unknown CMake command "pybind11_add_module".


-- Configuring incomplete, errors occurred!
```

**Comment by xinyazhang** [comment]
> @xinyazhang hi, ran into the issue trying to build today - would you mind updating the README or would you rather a PR?

@lhl Error message already added to CMakeLists.txt in a WIP branch commit (7d33da15a0029a32eb8d17e0cbf2000804317c1f) and will be included in 0.8b. Closing

**Comment by githust66** [comment]
> > aotriton-0.7.3b-manylinux_2_28_x86_64-rocm6.2-shared.tar ，Does this package support Navi GPUs?
> 
> Yes it supports, but the performance is very questionable and not recommended unless you absolutely want it.
> 
> > How should I use this package, or can I compile it from source by myself?
> 
> If you only want Navi support, you can use PyTorch nightly directly. 0.7.3b has no fundamental change compared with 0.7b PyTorch's Navi support needs to be enabled by `TORCH_ROCM_AOTRITON_ENABLE_EXPERIMENTAL=1`

Do you mean that Navi GPUs are still not supported?，What exactly should I do, my environment is
![image](https://github.com/user-attachments/assets/0bbf70d1-10a8-42b8-ab0e-5d04c9313d8a)
